README.md

```diff
@@ -4,6 +4,113 @@ Motoon Mentorship Program Team Educational Project
 
 <img src='Screenshots/cover.png' width=100% alt='Cover Photo'>
 
+## DI Container 💉
+**DI container** is like a central hub that stores and manages the dependencies.
+
+We are using [**Factory Package**](https://github.com/hmlongco/Factory) as a Dependency Injection Cantainer.
+
+### How to use it:
+Here's a simple dependency registration that returns a service that conforms to `MyServiceType`.
+
+```swift
+extension Container {
+    var myService: Factory<MyServiceType> { 
+        Factory(self) { MyService() }
+    }
+}
+```
+
+Injecting an instance of our service is straightforward, you can do it in two ways:
+
+**1**
+
+```swift
+class ContentViewModel: ObservableObject {
+    @Injected(\.myService) private var myService
+    ...
+}
+```
+**2**
+```swift
+class ContentViewModel: ObservableObject {
+    private let myService = Container.shared.myService()
+    ...
+}
+```
+### Example:
+Here's our **Model:**
+
+```swift
+struct Meal {
+    let name: String
+    let ingrediants: [String]  
+}
+```
+
+Registar a `Model` Object within the container
+
+```swift
+extension Container {
+    var meal: Factory<Meal> {
+        Factory(self) { Meal(name: "Omlet", ingrediants: ["Salt", "Oil", "2 Eggs"]) }
+    }
+```
+
+**ViewModel:**
+
+```swift
+final class HomeViewModel {
+    // Injecting an instance of Meal
+    @Injected(\.meal) private var meal
+    
+    private func numberOfIngrediants() -> Int {
+        meal.ingrediants.count   // -> 3
+    }
+}
+```
+
+Registar a `ViewModel` Object within the container
+
+```swift
+extension Container {
+    var meal: Factory<Meal> {
+        Factory(self) { Meal(name: "Omlet", ingrediants: ["Salt", "Oil", "2 Eggs"]) }
+        
+
+    var homeViewModel: Factory<HomeViewModel> {
+        Factory(self) { HomeViewModel()
+        }
+    }
+    }
+```
+    
+
+**View:**
+
+```swift
+class HomeViewController: UIViewController {
+    // Injecting an instance of HomeViewModel
+    private let homeViewModel = Container.shared.homeViewModel()
+    
+}
+```
+
+**Dependency Registration:**
+
+
+```swift
+extension Container {
+    var meal: Factory<Meal> {
+        Factory(self) { Meal(name: "Omlet", ingrediants: ["Salt", "Oil", "2 Eggs"]) }
+    }
+    
+    var homeViewModel: Factory<HomeViewModel> {
+        Factory(self) { HomeViewModel()
+        }
+    }
+}
+```
```

Comment:
```suggestion
```

This part has a lot of duplications and few invalid points like we don't normally register the model for reusability. I suggest to keep the first section which is how to use it until we agree on the architecture. 

Thanks for your efforts